php-frontend/src/main/java/org/sonar/php/tree/impl/statement/NamespaceStatementTreeImpl.java

```diff
@@ -0,0 +1,122 @@
+ */
+package org.sonar.php.tree.impl.statement;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterators;
+import org.sonar.php.tree.impl.PHPTree;
+import org.sonar.php.tree.impl.lexical.InternalSyntaxToken;
+import org.sonar.plugins.php.api.tree.Tree;
+import org.sonar.plugins.php.api.tree.declaration.NamespaceNameTree;
+import org.sonar.plugins.php.api.tree.statement.NamespaceStatementTree;
+import org.sonar.plugins.php.api.tree.statement.StatementTree;
+import org.sonar.plugins.php.api.visitors.TreeVisitor;
+
+import javax.annotation.Nullable;
+import java.util.Iterator;
+import java.util.List;
+
+public class NamespaceStatementTreeImpl extends PHPTree implements NamespaceStatementTree {
+
+  private static final Kind KIND = Kind.NAMESPACE_STATEMENT;
+
+  private final InternalSyntaxToken namespaceToken;
+  private final NamespaceNameTree namespaceName;
+  private final InternalSyntaxToken openCurlyBrace;
+  private final List<StatementTree> statements;
+  private final InternalSyntaxToken closeCurlyBrace;
+  private final InternalSyntaxToken eosToken;
+
+  public NamespaceStatementTreeImpl(InternalSyntaxToken namespaceToken, NamespaceNameTree namespaceName, InternalSyntaxToken eosToken) {
+    this.namespaceToken = namespaceToken;
+    this.namespaceName = namespaceName;
+    this.eosToken = eosToken;
+
+    this.openCurlyBrace = null;
+    this.statements = ImmutableList.of();
+    this.closeCurlyBrace = null;
+  }
+
+  public NamespaceStatementTreeImpl(
+      InternalSyntaxToken namespaceToken, @Nullable NamespaceNameTree namespaceName,
+      InternalSyntaxToken openCurlyBrace, List<StatementTree> statements, InternalSyntaxToken closeCurlyBrace
+  ) {
+    this.namespaceToken = namespaceToken;
+    this.namespaceName = namespaceName;
+    this.openCurlyBrace = openCurlyBrace;
+    this.statements = statements;
+    this.closeCurlyBrace = closeCurlyBrace;
+
+    this.eosToken = null;
+  }
+
+  @Override
+  public InternalSyntaxToken namespaceToken() {
+    return namespaceToken;
+  }
+
+  @Nullable
+  @Override
+  public NamespaceNameTree namespaceName() {
+    return namespaceName;
+  }
+
+  @Nullable
+  @Override
+  public InternalSyntaxToken openCurlyBrace() {
+    return openCurlyBrace;
+  }
+
+  @Override
+  public List<StatementTree> statements() {
+    return statements;
+  }
+
+  @Nullable
+  @Override
+  public InternalSyntaxToken closeCurlyBrace() {
+    return closeCurlyBrace;
+  }
+
+  @Nullable
+  @Override
+  public InternalSyntaxToken eosToken() {
+    return eosToken;
+  }
+
+  @Override
+  public Kind getKind() {
+    return KIND;
+  }
+
+  @Override
+  public Iterator<Tree> childrenIterator() {
+    return Iterators.concat(
+        Iterators.forArray(namespaceToken, namespaceName, openCurlyBrace),
+        statements.iterator(),
+        Iterators.forArray(closeCurlyBrace, eosToken)
+    );
+  }
+
+  @Override
+  public void accept(TreeVisitor visitor) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1186)